cdk/lib/index.ts

```diff
@@ -0,0 +1,302 @@
+  RemovalPolicy,
+  SecretValue,
+} from "@aws-cdk/core";
+import { Certificate } from "@aws-cdk/aws-certificatemanager";
+import type { GuStackProps } from "@guardian/cdk/lib/constructs/core/stack";
+import {
+  GuDnsRecordSet,
+  RecordType,
+} from "@guardian/cdk/lib/constructs/dns/dns-records";
+import { GuStack } from "@guardian/cdk/lib/constructs/core/stack";
+import { AccessScope, GuPlayApp } from "@guardian/cdk";
+import {
+  GuGetS3ObjectsPolicy,
+  GuPutCloudwatchMetricsPolicy,
+} from "@guardian/cdk/lib/constructs/iam";
+import { GuSecurityGroup, GuVpc } from "@guardian/cdk/lib/constructs/ec2";
+import { InstanceType, Port, SubnetType } from "@aws-cdk/aws-ec2";
+import { GuS3Bucket } from "@guardian/cdk/lib/constructs/s3";
+import {
+  AllowedMethods,
+  CacheCookieBehavior,
+  CacheHeaderBehavior,
+  CachePolicy,
+  CacheQueryStringBehavior,
+  Distribution,
+  OriginProtocolPolicy,
+} from "@aws-cdk/aws-cloudfront";
+import { LoadBalancerV2Origin } from "@aws-cdk/aws-cloudfront-origins";
+import {
+  Alarm,
+  ComparisonOperator,
+  Metric,
+  TreatMissingData,
+} from "@aws-cdk/aws-cloudwatch";
+import { GuDatabaseInstance } from "@guardian/cdk/lib/constructs/rds";
+import {
+  Credentials,
+  DatabaseInstanceEngine,
+  PostgresEngineVersion,
+  StorageType,
+  SubnetGroup,
+} from "@aws-cdk/aws-rds";
+import { GuArnParameter, GuParameter } from "@guardian/cdk/lib/constructs/core";
+
+export class Typerighter extends GuStack {
+  constructor(scope: App, id: string, props: GuStackProps) {
+    super(scope, id, props);
+
+    const parameters = {
+      MasterDBUsername: new GuParameter(this, "MasterDBUsername", {
+        description: "Master DB username",
+        default: "rule_manager",
+        type: "String",
+      }),
+      CheckerCertificate: new GuArnParameter(
+        this,
+        "CheckerCloudfrontCertificate",
+        {
+          description:
+            "The ARN of the certificate for the checker service Cloudfront distribution",
+        }
+      ),
+    };
+
+    const pandaAuthPolicy = new GuGetS3ObjectsPolicy(this, "PandaAuthPolicy", {
+      bucketName: "pan-domain-auth-settings",
+    });
+
+    const checkerDomainCODE = "checker.typerighter.code.dev-gutools.co.uk";
+    const checkerDomainPROD = "checker.typerighter.gutools.co.uk";
+    const managerDomainCODE = "manager.typerighter.code.dev-gutools.co.uk";
+    const managerDomainPROD = "manager.typerighter.gutools.co.uk";
+
+    const stageLookup = new CfnMapping(this, "LowercaseStageLookup", {
+      mapping: {
+        lowercase: {
+          PROD: "prod",
+          CODE: "code",
+        },
+        checkerDomain: {
+          PROD: checkerDomainPROD ,
+          CODE: checkerDomainCODE,
+        },
+        managerDomain: {
+          PROD: managerDomainPROD,
+          CODE: managerDomainCODE,
+        },
+      },
+    });
+
+    const typerighterBucketName = `typerighter-${stageLookup.findInMap(
+      "lowercase",
+      this.stage
+    )}`;
```

Comment:
I'm increasingly of the opinion that in this scenario we should have a single bucket that is sharded by stage, rather than a bucket per stage (akin to an account's artifact bucket).

Would love to get your opinion on that/hear why you have a bucket per stage.

Reply:
cc @jfsoul @jacobwinch @nicl as we were discussing this recently.

Reply:
Makes sense, and an easy change to make.